src/main/java/bio/terra/cli/apps/DockerUtils.java

```diff
@@ -0,0 +1,244 @@
+    List<Bind> bindMountsObj = new ArrayList<>();
+    for (Map.Entry<Path, Path> bindMount : bindMounts.entrySet()) {
+      File localDirectory = bindMount.getValue().toFile();
+      if (!localDirectory.exists() || !localDirectory.isDirectory()) {
+        throw new SystemException(
+            "Bind mount does not specify a local directory: " + localDirectory.getAbsolutePath());
+      }
+      bindMountsObj.add(
+          new Bind(localDirectory.getAbsolutePath(), new Volume(bindMount.getKey().toString())));
+    }
+
+    // create the container and start it
+    CreateContainerCmd createContainerCmd =
+        dockerClient
+            .createContainerCmd(imageId)
+            .withCmd("bash", "-c", command)
+            .withEnv(envVarsStr)
+            .withHostConfig(HostConfig.newHostConfig().withBinds(bindMountsObj))
+            .withAttachStdout(true)
+            .withAttachStderr(true);
+    if (workingDir != null) {
+      createContainerCmd.withWorkingDir(workingDir);
+    }
+    try {
+      CreateContainerResponse container = createContainerCmd.exec();
+      dockerClient.startContainerCmd(container.getId()).exec();
+      return container.getId();
+    } catch (RuntimeException rtEx) {
+      throw wrapExceptionIfDockerConnectionFailed(rtEx);
+    }
+  }
+
+  /**
+   * Block until the Docker container exits, then return its status code.
+   *
+   * @param containerId id of the container
+   */
+  public Integer waitForDockerContainerToExit(String containerId) {
+    WaitContainerResultCallback waitContainerResultCallback = new WaitContainerResultCallback();
+    try {
+      WaitContainerResultCallback exec =
+          dockerClient.waitContainerCmd(containerId).exec(waitContainerResultCallback);
+      return exec.awaitStatusCode();
+    } catch (RuntimeException rtEx) {
+      throw wrapExceptionIfDockerConnectionFailed(rtEx);
+    }
+  }
+
+  /**
+   * Delete the Docker container.
+   *
+   * @param containerId id of the container
+   */
+  public void deleteDockerContainer(String containerId) {
+    try {
+      dockerClient.removeContainerCmd(containerId).exec();
+    } catch (RuntimeException rtEx) {
+      throw wrapExceptionIfDockerConnectionFailed(rtEx);
+    }
+  }
+
+  /**
+   * Read the Docker container logs and write them to standard out.
+   *
+   * @param containerId id of the container
+   */
+  public void outputLogsForDockerContainer(String containerId) {
+    try {
+      dockerClient
+          .logContainerCmd(containerId)
+          .withStdOut(true)
+          .withStdErr(true)
+          .withFollowStream(true)
+          .withTailAll()
+          .exec(new LogContainerTestCallback());
+    } catch (RuntimeException rtEx) {
+      throw wrapExceptionIfDockerConnectionFailed(rtEx);
+    }
+  }
+
+  /** Helper class for reading Docker container logs into a string. */
+  private static class LogContainerTestCallback extends ResultCallback.Adapter<Frame> {
```

Comment:
nit: maybe `LogCallback`? Not sure what's `Test` about this.

Reply:
Agreed, I think that's leftover from some debugging when I originally wrote this. Changed to `LogContainerCommandCallback`.